pkg/api/v1beta1/dynakube/validation/oneagent.go

```diff
@@ -143,15 +147,20 @@ func conflictingHostGroupSettings(_ context.Context, _ *Validator, dk *dynakube.
 	return ""
 }
 
-func validateOneAgentVersionIsSemVerCompliant(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+func isOneAgentVersionValid(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
 	agentVersion := dk.CustomOneAgentVersion()
 	if agentVersion == "" {
 		return ""
 	}
 
-	version := "v" + agentVersion
-	if !(semver.IsValid(version) && semver.Prerelease(version) == "" && semver.Build(version) == "" && len(strings.Split(version, ".")) == 3) {
-		return "Only semantic versions in the form of major.minor.patch (e.g. 1.0.0) are allowed!"
+	_, err := dtversion.ToSemver(agentVersion)
+	if err != nil {
+		return versionInvalidMessage
+	}
```

Comment:
is this necessary? if you have the regexp?

Reply:
In there we do semver.isValid() so its useful to check this too.